source/common/http/header_map_impl.cc

```diff
@@ -178,6 +178,50 @@ template <> bool HeaderMapImpl::HeaderList::isPseudoHeader(const LowerCaseString
   return key.get().c_str()[0] == ':';
 }
 
+bool HeaderMapImpl::HeaderList::maybeMakeMap() {
+  if (lazy_map_.empty()) {
+#if HEADER_MAP_SIZE_THRESHOLD != 0
+    if (headers_.size() < HEADER_MAP_SIZE_THRESHOLD) {
+      return false;
+    }
+#endif
+    // add all entries from the list into the map
+    for (auto node = headers_.begin(); node != headers_.end(); ++node) {
+      HeaderNodeVector& v = lazy_map_[node->key().getStringView()];
+      v.push_back(node);
+    }
+  }
+  return true;
+}
+
+size_t HeaderMapImpl::HeaderList::remove(absl::string_view key) {
+  size_t removed_bytes = 0;
+  if (maybeMakeMap()) {
+    auto iter = lazy_map_.find(key);
+    if (iter != lazy_map_.end()) {
+      // erase from the map, and all same key entries from the list
```

Comment:
Begin sentences with upper case, end with a period, here & elsewhere in the PR.

Reply:
Done.